src/common.js

```diff
@@ -25,17 +26,32 @@ function log() {
 }
 exports.log = log;
 
+var DEFAULT_RETURN_CODE = 1;
+
 // Shows error message. Throws unless _continue or config.fatal are true
-function error(msg, _continue) {
+function error(msg, _code, _continue) {
+  // Adjust values if `_continue` is passed in but `_code` isn't
+  if (typeof _code === 'boolean') {
+    _continue = _code;
+    _code = DEFAULT_RETURN_CODE;
+  }
+
+  if (typeof _code !== 'number')
+    _code = DEFAULT_RETURN_CODE;
+
   if (state.error === null)
     state.error = '';
   state.error += state.currentCmd + ': ' + msg + '\n';
 
+  // Override previous return code
+  if (state.errorCode === 0 || _code !== DEFAULT_RETURN_CODE)
```

Comment:
I wrote it like this for two reasons:

 1. This prefers non-default return codes. If a command has two errors (and `fatal` === false, so it doesn't exit yet), and one is a non-default return code, then it prefers that value overall (which probably tells the user more useful information).
 2. This gives the user a way to send warning messages (output like an error, but don't change the status code) by providing `_code === 0`

I thought about changing it to `if (state.errorCode === 0 || state.errorCode === DEFAULT_RETURN_CODE)` instead, but that would allow `error('...', 0)` to wipe out a previous non-zero code, which it probably shouldn't do.

What do you think?